### ISSUE TITLE
 Validate the initial configuration of wazuh_table and pf to the .conf file

### ISSUE THREAD

**Issue by QU3B1M** [wazuh/wazuh-qa#3330]
| Target version | Related issue | Related PR |
|--------------------|--------------------|-----------------|
|    4.5          |[#14073](https://github.com/wazuh/wazuh/issues/14073)|[#14962](https://github.com/wazuh/wazuh/pull/14962)|

## Description

This update solves the issue [#14073](https://github.com/wazuh/wazuh/issues/14073), the proposed solution was that when a wazuh_fwtable does not exist, it is created , the configurations are added to the pf.conf and rc.conf files and pf rules are reload. Also added an alert when the firewall is disabled, it is verified if the firewall is down and if so, it is written in the active response log in order to be able to generate the alert, for this the rule (658) is used.

## Proposed checks

- <details><summary>Scenario Outline 1 - PoC environment and raised alert validation.</summary>
    
    ```Gherkin
    Given a wazuh-manager and a "<agent_os>" wazuh-agent.
    And the agent's firewall "<firewall_status>".
    And the environment configured as in https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html.
    When connect to the agent system by ssh from the attacker machine.
    Then the attacker's IP is "<IP_result>".
    And the block-malicious-actor alert is raised.
    And the alert firewall-disabled alert is "<firewall_alert_status>".

    Examples:
      
      | agent_os           | firewall_status | IP_result  | firewall_alert_status |
      |--------------------|-----------------|------------|-----------------------|
      | Macos              | enabled         | blocked    | not raised            |
      | Macos              | disabled        | not blocked| raised                |
      | FreeBSD (optional) | enabled         | blocked    | not raised            |
      | FreeBSD (optional) | disabled        | not blocked| raised                |
      | OpenBSD (optional) | enabled         | blocked    | not raised            |
      | OpenBSD (optional) | disabled        | not blocked| raised                |
    ```
</details>

- <details><summary>Scenario Outline 2 - Firewall's files validation.</summary>

    ```Gherkin
    Given a wazuh-manager and a not started "<agent_os>" wazuh-agent.
    And the agent's firewal "<firewall_status>".
    And the file /etc/pf.conf is "<pf_conf_initial_status>".
    When start the wazuh-agent.
    Then the file /etc/pf.conf is "<pf_conf_final_status>".

    Examples:
    
      | agent_os           | firewall_status | pf_conf_initial_status | pf_conf_final_status |
      |--------------------|-----------------|------------------------|----------------------|
      | Macos              | enabled         | not created            | created              |
      | Macos              | enabled         | created                | created              |
      | Macos              | disabled        | not created            | not created          |
      | Macos              | disabled        | created                | created              |
      | FreeBSD (optional) | enabled         | not created            | created              |
      | FreeBSD (optional) | enabled         | created                | created              |
      | FreeBSD (optional) | disabled        | not created            | not created          |
      | FreeBSD (optional) | disabled        | created                | created              |
      | OpenBSD (optional) | enabled         | not created            | created              |
      | OpenBSD (optional) | enabled         | created                | created              |
      | OpenBSD (optional) | disabled        | not created            | not created          |
      | OpenBSD (optional) | disabled        | created                | created              |
   ```

</details>

### Tasks
- [x] Fresh install
- [x] Upgrade

## Expected results

When an `active-response` alert is raised and the firewall is *disabled* an alert should be generated to inform the user that as the firewall is down, the rule created by `active-response` wont have any effect.\
In the case that the firewall is *enabled* the `active-response` firewall rule has effect and blocks the attacker's IP

## Configuration and considerations

<details><summary>Manager <code>ossec.conf</code></summary>
  
  ```XML
  <ossec_config>

    <active-response>
      <command>firewall-drop</command>
      <location>local</location>
      <rules_id>100100</rules_id>
      <timeout>60</timeout>
    </active-response>

    <ruleset>
      <!-- Default ruleset -->
      <decoder_dir>ruleset/decoders</decoder_dir>
      <rule_dir>ruleset/rules</rule_dir>
      <rule_exclude>0215-policy_rules.xml</rule_exclude>
      <list>etc/lists/audit-keys</list>
      <list>etc/lists/blacklist-alienvault</list>
      <!-- User-defined ruleset -->
      <decoder_dir>etc/decoders</decoder_dir>
      <rule_dir>etc/rules</rule_dir>
    </ruleset>

    <active-response>
      <command>firewall-drop</command>
      <location>local</location>
      <rules_id>100100</rules_id>
      <timeout>60</timeout>
    </active-response>
  </ossec_config>
  ```
</details>

<details><summary>Agent <code>ossec.conf</code></summary>

  ```XML
  <ossec_config>
    <localfile>
      <log_format>apache</log_format>
      <location>/var/log/apache2/access_log</location>
    </localfile>
  </ossec_config>
  ```
</details>

**Comment by QU3B1M** [comment]
## Review data

| Tester | PR commit               | 
|--------|-------------------------|
| @QU3B1M   |  [c4abd1b](https://github.com/wazuh/wazuh/pull/14962/commits/c4abd1b0682ce40a85c64c40604bb2db55260a88)   | 

### Testing environment

| OS | OS version | Deployment                                    | Image/AMI | Notes |
|----|------------|-----------------------------------------------|-----------|-------|
|  Ubuntu  | 20.04 | ` LOCAL \|  Vagrant` | [ubuntu/focal64](https://app.vagrantup.com/ubuntu/boxes/focal64)
|  MacOs | mojave | ` LOCAL \|  Vagrant` | [danimaetrix/macOS-mojave](https://app.vagrantup.com/danimaetrix/boxes/macOS-mojave) ||

### Tested packages

| `wazuh-manager` | `wazuh-agent` | 
|-----------------|---------------|    
|[deb](https://packages-dev.wazuh.com/warehouse/pullrequests/4.5/deb/var/wazuh-agent_4.5.0-0.commitc4abd1b_amd64.deb)|[macos](https://packages-dev.wazuh.com/warehouse/pullrequests/4.5/macos/wazuh-agent-4.5.0-0.commitc4abd1b.pkg)|       

### Status

- [x] In progress @QU3B1M 
- [ ] Pending Review
- [ ] QA team leader approved @damarisg 
- [ ] QA manager approved @jmv74211 
- [ ] Development team leader approved @TomasTurina 

### Conclusion

**Comment by QU3B1M** [comment]
# Update 20/01/2023

**Comment by QU3B1M** [comment]
# Testing results :white_circle: 

## Upgrade

  <details><summary>PoC Alert is generated and IP correctly blocked</summary>

  - <details><summary>When the firewall is enabled the alert is generated and IP blocked</summary>

    1. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    2. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    3. Follow this [POC guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html) and apply the configuration to the manager and agent `ossec.conf`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. From the attacker system (a third VM) connect to the vicitm's system (macOS agent)
    6. Check the `block-malicious-actor` alert is generated
        `Example`

        ```JSON
        {"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":4,"mail":false,"groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"}

        {"timestamp":"2022-09-17T14:20:08.605-0500","rule":{"level":3,"description":"Host Blocked by firewall-drop Active Response","id":"651","firedtimes":18,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gpg13":["4.13"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315307","full_log":"2022/09/17 14:20:08 active-response/bin/firewall-drop: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-09-17T14:20:08.552-0500\",\"rule\":{\"level\":10,\"description\":\"IP address found in AlienVault reputation database.\",\"id\":\"100100\",\"firedtimes\":4,\"mail\":false,\"groups\":[\"attack\"]},\"agent\":{\"id\":\"006\",\"name\":\"Vagrants-MacBook-Pro-Bck.local\",\"ip\":\"FE80:0000:0000:0000:1876:67B9:B5D4:0765\"},\"manager\":{\"name\":\"thejbte-ASUS-TUF-A15\"},\"id\":\"1663442408.1315007\",\"full_log\":\"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \\\"GET / HTTP/1.1\\\" 304 -\",\"decoder\":{\"name\":\"web-accesslog\"},\"data\":{\"protocol\":\"GET\",\"srcip\":\"192.168.56.66\",\"id\":\"304\",\"url\":\"/\"},\"location\":\"/var/log/apache2/access_log\"},\"program\":\"active-response/bin/firewall-drop\"}}","decoder":{"name":"ar_log_json"},"data":{"version":"1","origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":"10","description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":"4","mail":"false","groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"},"program":"active-response/bin/firewall-drop"}},"location":"/Library/Ossec/logs/active-responses.log"}
        ```
    7. From the attacker system try to connect to the agent's system by ssh and check its IP is blocked

    </details>

  - <details><summary>When the firewall is disabled the alert is generated but IP is not blocked</summary>

    1. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    2. Ensure the macOS agent's firewall is disabled 
        > if not turn it off with: `pfctl -d`
    3. Follow this [POC guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html) and apply the configuration to the manager and agent `ossec.conf`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. From the attacker system (a third VM) connect to the vicitm's system (macOS agent)
    6. Check the `block-malicious-actor` alert is generated
        `Example`

        ```JSON
        {"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":4,"mail":false,"groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"}

        {"timestamp":"2022-09-17T14:20:08.605-0500","rule":{"level":3,"description":"Host Blocked by firewall-drop Active Response","id":"651","firedtimes":18,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gpg13":["4.13"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315307","full_log":"2022/09/17 14:20:08 active-response/bin/firewall-drop: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-09-17T14:20:08.552-0500\",\"rule\":{\"level\":10,\"description\":\"IP address found in AlienVault reputation database.\",\"id\":\"100100\",\"firedtimes\":4,\"mail\":false,\"groups\":[\"attack\"]},\"agent\":{\"id\":\"006\",\"name\":\"Vagrants-MacBook-Pro-Bck.local\",\"ip\":\"FE80:0000:0000:0000:1876:67B9:B5D4:0765\"},\"manager\":{\"name\":\"thejbte-ASUS-TUF-A15\"},\"id\":\"1663442408.1315007\",\"full_log\":\"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \\\"GET / HTTP/1.1\\\" 304 -\",\"decoder\":{\"name\":\"web-accesslog\"},\"data\":{\"protocol\":\"GET\",\"srcip\":\"192.168.56.66\",\"id\":\"304\",\"url\":\"/\"},\"location\":\"/var/log/apache2/access_log\"},\"program\":\"active-response/bin/firewall-drop\"}}","decoder":{"name":"ar_log_json"},"data":{"version":"1","origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":"10","description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":"4","mail":"false","groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"},"program":"active-response/bin/firewall-drop"}},"location":"/Library/Ossec/logs/active-responses.log"}
        ```
    7. Check the `firewall is disabled` alert is generated
        `Example`
        ```JSON
        {"timestamp":"2022-09-21T13:38:59.529-0500","rule":{"level":5,"description":"Active response related to macos firewall has been activated, but may not have an effect because the firewall is inactive","id":"658","firedtimes":2,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"002","name":"DESKTOP-QCL0GB6","ip":"FE80:0000:0000:0000:44D6:6AA2:2014:D817"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663785539.1153467","full_log":"2022/09/21 18:38:57 active-response/bin/netsh.exe: {\"message\":\"Active response may not have an effect\",\"firewall\":{\"profile1\":\"FIREWALL_DOMAIN\",\"status1\":\"inactive\",\"profile2\":\"FIREWALL_PRIVATE\",\"status2\":\"active\",\"profile3\":\"FIREWALL_PUBLIC\",\"status3\":\"inactive\"},\"status\":\"inactive\",\"script\":\"netsh\"}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"status":"inactive","message":"Active response may not have an effect","firewall":{"profile1":"FIREWALL_DOMAIN","status1":"inactive","profile2":"FIREWALL_PRIVATE","status2":"active","profile3":"FIREWALL_PUBLIC","status3":"inactive"},"script":"netsh"},"location":"active-response\\active-responses.log"}
        ```
    8. From the attacker system try to connect to the agent's system by ssh and check its IP is not blocked
    </details>


</details>

  <details><summary>MacOs firewall's files are created and updated when they should</summary>

  - <details><summary>When the firewall is enabled and the file <code>/etc/rc.conf</code> does not exist it's created </summary>

    1. Ensure the file `/etc/rc.conf` does not exist and the file `/etc/pf.conf` is not updated.
        > If it exists, delete it `rm -rf /etc/rc.conf`

        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` is now created and its content is correct
        `Example`
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
    6. Check the file `/etc/pf.conf` is updated correctly
        `Example` *(added to the file)* 
        ```
        ...
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>
    
  - <details><summary>When the firewall is disabled and the file <code>/etc/rc.conf</code> does not exist it wont be created</summary>

    1. Ensure the file `/etc/rc.conf` does not exist and the file `/etc/pf.conf` is not updated.
        > If it exists, delete it `rm -rf /etc/rc.conf`

        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is disabled 
        > if not turn it off with: `pfctl -d`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` is not created
        `cat /etc/rc.conf`
    6. Check the file `/etc/pf.conf` is not updated
    </details>

  - <details><summary>When the firewall is enabled and the file <code>/etc/rc.conf</code> already exist and <code>/etc/pf.conf</code> is not updated it gets updated correctly</summary>

    1. Ensure the file `/etc/rc.conf` exist with correct content and the file `/etc/pf.conf` is not updated.
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` content is correct
        `Example`
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
    6. Check the file `/etc/pf.conf` is updated correctly
        `Example` *(added to the file)* 
        ```
        ...
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>When the firewall is enabled and the file <code>/etc/rc.conf</code> already exist and <code>/etc/pf.conf</code> is updated correctly, nothing happens</summary>

    1. Ensure the file `/etc/rc.conf` exist with correct content and the file `/etc/pf.conf` is updated.
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          table <wazuh_fwtable> persist #wazuh_fwtable
          block in quick from <wazuh_fwtable> to any
          block out quick from any to <wazuh_fwtable>
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` content is still correct
    6. Check the file `/etc/pf.conf` is still correct

    </details>
</details>

## Fresh install

  <details><summary>PoC Alert is generated and IP correctly blocked</summary>

  - <details><summary>When the firewall is enabled the alert is generated and IP blocked</summary>

    1. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    2. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    3. Follow this [POC guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html) and apply the configuration to the manager and agent `ossec.conf`
    4. Start the `wazuh-manager` and `wazuh-agent`
      `ubuntu manager`
      ```
      /var/ossec/bin/wazuh-control start
      ```
      `macos agent`
      ```
      /Library/Ossec/bin/wazuh-control start
      ```
    5. From the attacker system (a third VM) connect to the vicitm's system (macOS agent)
    6. Check the `block-malicious-actor` alert is generated

        ```JSON
        {"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":4,"mail":false,"groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"}

        {"timestamp":"2022-09-17T14:20:08.605-0500","rule":{"level":3,"description":"Host Blocked by firewall-drop Active Response","id":"651","firedtimes":18,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gpg13":["4.13"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315307","full_log":"2022/09/17 14:20:08 active-response/bin/firewall-drop: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-09-17T14:20:08.552-0500\",\"rule\":{\"level\":10,\"description\":\"IP address found in AlienVault reputation database.\",\"id\":\"100100\",\"firedtimes\":4,\"mail\":false,\"groups\":[\"attack\"]},\"agent\":{\"id\":\"006\",\"name\":\"Vagrants-MacBook-Pro-Bck.local\",\"ip\":\"FE80:0000:0000:0000:1876:67B9:B5D4:0765\"},\"manager\":{\"name\":\"thejbte-ASUS-TUF-A15\"},\"id\":\"1663442408.1315007\",\"full_log\":\"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \\\"GET / HTTP/1.1\\\" 304 -\",\"decoder\":{\"name\":\"web-accesslog\"},\"data\":{\"protocol\":\"GET\",\"srcip\":\"192.168.56.66\",\"id\":\"304\",\"url\":\"/\"},\"location\":\"/var/log/apache2/access_log\"},\"program\":\"active-response/bin/firewall-drop\"}}","decoder":{"name":"ar_log_json"},"data":{"version":"1","origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":"10","description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":"4","mail":"false","groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"},"program":"active-response/bin/firewall-drop"}},"location":"/Library/Ossec/logs/active-responses.log"}
        ```
    7. From the attacker system try to connect to the agent's system by ssh and check its IP is blocked

    </details>

  - <details><summary>When the firewall is disabled the alert is generated but IP is not blocked</summary>

    1. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    2. Ensure the macOS agent's firewall is disabled 
        > if not turn it off with: `pfctl -d`
    3. Follow this [POC guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html) and apply the configuration to the manager and agent `ossec.conf`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. From the attacker system (a third VM) connect to the vicitm's system (macOS agent)
    6. Check the `block-malicious-actor` alert is generated
        `Example`

        ```JSON
        {"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":4,"mail":false,"groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"}

        {"timestamp":"2022-09-17T14:20:08.605-0500","rule":{"level":3,"description":"Host Blocked by firewall-drop Active Response","id":"651","firedtimes":18,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gpg13":["4.13"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315307","full_log":"2022/09/17 14:20:08 active-response/bin/firewall-drop: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-09-17T14:20:08.552-0500\",\"rule\":{\"level\":10,\"description\":\"IP address found in AlienVault reputation database.\",\"id\":\"100100\",\"firedtimes\":4,\"mail\":false,\"groups\":[\"attack\"]},\"agent\":{\"id\":\"006\",\"name\":\"Vagrants-MacBook-Pro-Bck.local\",\"ip\":\"FE80:0000:0000:0000:1876:67B9:B5D4:0765\"},\"manager\":{\"name\":\"thejbte-ASUS-TUF-A15\"},\"id\":\"1663442408.1315007\",\"full_log\":\"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \\\"GET / HTTP/1.1\\\" 304 -\",\"decoder\":{\"name\":\"web-accesslog\"},\"data\":{\"protocol\":\"GET\",\"srcip\":\"192.168.56.66\",\"id\":\"304\",\"url\":\"/\"},\"location\":\"/var/log/apache2/access_log\"},\"program\":\"active-response/bin/firewall-drop\"}}","decoder":{"name":"ar_log_json"},"data":{"version":"1","origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":"10","description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":"4","mail":"false","groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"},"program":"active-response/bin/firewall-drop"}},"location":"/Library/Ossec/logs/active-responses.log"}
        ```
    7. Check the `firewall is disabled` alert is generated
        `Example`
        ```JSON
        {"timestamp":"2022-09-21T13:38:59.529-0500","rule":{"level":5,"description":"Active response related to macos firewall has been activated, but may not have an effect because the firewall is inactive","id":"658","firedtimes":2,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"002","name":"DESKTOP-QCL0GB6","ip":"FE80:0000:0000:0000:44D6:6AA2:2014:D817"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663785539.1153467","full_log":"2022/09/21 18:38:57 active-response/bin/netsh.exe: {\"message\":\"Active response may not have an effect\",\"firewall\":{\"profile1\":\"FIREWALL_DOMAIN\",\"status1\":\"inactive\",\"profile2\":\"FIREWALL_PRIVATE\",\"status2\":\"active\",\"profile3\":\"FIREWALL_PUBLIC\",\"status3\":\"inactive\"},\"status\":\"inactive\",\"script\":\"netsh\"}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"status":"inactive","message":"Active response may not have an effect","firewall":{"profile1":"FIREWALL_DOMAIN","status1":"inactive","profile2":"FIREWALL_PRIVATE","status2":"active","profile3":"FIREWALL_PUBLIC","status3":"inactive"},"script":"netsh"},"location":"active-response\\active-responses.log"}
        ```
    8. From the attacker system try to connect to the agent's system by ssh and check its IP is not blocked
    </details>

</details>

  <details><summary>MacOs firewall's files are created and updated when they should</summary>

  - <details><summary>When the firewall is enabled and the file <code>/etc/rc.conf</code> does not exist it's created </summary>

    1. Ensure the file `/etc/rc.conf` does not exist and the file `/etc/pf.conf` is not updated.
        > If it exists, delete it `rm -rf /etc/rc.conf`

        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` is now created and its content is correct
        `Example`
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
    6. Check the file `/etc/pf.conf` is updated correctly
        `Example` *(added to the file)* 
        ```
        ...
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>
    
  - <details><summary>When the firewall is disabled and the file <code>/etc/rc.conf</code> does not exist it wont be created</summary>

    1. Ensure the file `/etc/rc.conf` does not exist and the file `/etc/pf.conf` is not updated.
        > If it exists, delete it `rm -rf /etc/rc.conf`

        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is disabled 
        > if not turn it off with: `pfctl -d`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` is not created
        `cat /etc/rc.conf`
    6. Check the file `/etc/pf.conf` is not updated
    </details>

  - <details><summary>When the firewall is enabled and the file <code>/etc/rc.conf</code> already exist and <code>/etc/pf.conf</code> is not updated it gets updated correctly</summary>

    1. Ensure the file `/etc/rc.conf` exist with correct content and the file `/etc/pf.conf` is not updated.
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` content is correct
        `Example`
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
    6. Check the file `/etc/pf.conf` is updated correctly
        `Example` *(added to the file)* 
        ```
        ...
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>When the firewall is enabled and the file <code>/etc/rc.conf</code> already exist and <code>/etc/pf.conf</code> is updated correctly, nothing happens</summary>

    1. Ensure the file `/etc/rc.conf` exist with correct content and the file `/etc/pf.conf` is updated.
        ```
        pf_enable="YES"
        pf_rules="/etc/pf.conf"
        pflog_enable="YES"
        pflog_logfile="/var/log/pflog"
        ```
        <details><summary>Default <code>/etc/pf.conf</code> content</summary>

          ```
          #
          # Default PF configuration file.
          #
          # This file contains the main ruleset, which gets automatically loaded
          # at startup.  PF will not be automatically enabled, however.  Instead,
          # each component which utilizes PF is responsible for enabling and disabling
          # PF via -E and -X as documented in pfctl(8).  That will ensure that PF
          # is disabled only when the last enable reference is released.
          #
          # Care must be taken to ensure that the main ruleset does not get flushed,
          # as the nested anchors rely on the anchor point defined here. In addition,
          # to the anchors loaded by this file, some system services would dynamically 
          # insert anchors into the main ruleset. These anchors will be added only when
          # the system service is used and would removed on termination of the service.
          #
          # See pf.conf(5) for syntax.
          #

          #
          # com.apple anchor point
          #
          scrub-anchor "com.apple/*"
          nat-anchor "com.apple/*"
          rdr-anchor "com.apple/*"
          dummynet-anchor "com.apple/*"
          anchor "com.apple/*"
          load anchor "com.apple" from "/etc/pf.anchors/com.apple"
          table <wazuh_fwtable> persist #wazuh_fwtable
          block in quick from <wazuh_fwtable> to any
          block out quick from any to <wazuh_fwtable>
          ```
        </details>

    2. Install `wazuh-manager` (linux) and a `wazuh-agent` on macOS
    3. Ensure the macOS agent's firewall is enabled 
        > if not turn it on with: `pfctl -e`
    4. Start the `wazuh-manager` and `wazuh-agent`
    5. Check the file `/etc/rc.conf` content is still correct
    6. Check the file `/etc/pf.conf` is still correct

    </details>
</details>

**Comment by QU3B1M** [comment]
### Update 31/01/2023 - On Hold :clock1: 

This issue will be on hold to prioritize the 4.4.0 `agent_group` related

**Comment by QU3B1M** [comment]
### Update 15/02/2023

The tests cases were re-maked applying Gherkin language and re-validated, some other OS were added to test.

**Comment by QU3B1M** [comment]
### Update 22/03/2023
- Testing resumed

**Comment by QU3B1M** [comment]
## Fresh install

<details><summary>Scenario 1: PoC environment and raised alert validation</summary>

  > <details><summary>Gherkin scenario description</summary>
  >
  >    ```Gherkin
  >    Given a wazuh-manager and a "<agent_os>" wazuh-agent.
  >    And the agent's firewal "<firewall_status>".
  >    And the environment configured as in the POC: https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html.
  >    When connect to the agent system from the attacker machine.
  >    Then the attacker's IP is "<IP_result>".
  >    And the POC alert is raised.
  >    
  >    And the alert "Active Response may not have an effect" alert is "<firewall_alert_status>"
  >  
  >    Examples:
  >    
  >    | agent_os| firewall_status | IP_result  | firewall_alert_status |
  >    |---------|-----------------|------------|-----------------------|
  >    | Macos   | enabled         | blocked    | not raised            |
  >    | Macos   | disabled        | not blocked| raised                |
  >    | FreeBSD | enabled         | blocked    | not raised            |
  >    | FreeBSD | disabled        | not blocked| raised                |
  >    | OpenBSD | enabled         | blocked    | not raised            |
  >    | OpenBSD | disabled        | not blocked| raised                |
  >    ```
  >  </details>

  - <details><summary>Case 1 - Macos firewall enabled IP is blocked :green_circle:</summary>
    
    1. Ensure the firewall is enabled on the `macos` system.
        ```powershell
        pfctl -e
        ```
    2. Configure the POC environment from [this guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html).
    3. "Attack" the macos system from an third system.
    4. Check the `IP address found in AlienVault reputation database.` alert is raised.
        ```JSON
        egrep "IP address found" /var/ossec/logs/alerts/alerts.json

        {"timestamp":"2023-02-23T12:06:58.165+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":1,"mail":false,"groups":["attack"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677154018.18950","full_log":"192.168.56.12 - - [23/Feb/2023:06:06:59 -0600] \"GET / HTTP/1.1\" 200 45","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.12","id":"200","url":"/"},"location":"/private/var/log/apache2/access_log"}
        ```
    5. Check the `Active response may not have an effect` alert is **not** raised.
        ```JSON
        egrep "Active response may not" /var/ossec/logs/alerts/alerts.json
        ```

    </details>

  - <details><summary>Case 2 - Macos firewall disabled IP is blocked :green_circle:</summary>
    
    1. Ensure the firewall is disabled on the `macos` system.
        ```powershell
        pfctl -d
        ```
    2. Configure the POC environment from [this guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html).
    3. "Attack" the macos system from an third system.
    4. Check the `IP address found in AlienVault reputation database.` alert is raised.
        ```JSON
        egrep "IP address found" /var/ossec/logs/alerts/alerts.json

        {"timestamp":"2023-02-23T12:25:02.813+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":7,"mail":false,"groups":["attack"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677155102.42932","full_log":"192.168.56.12 - - [23/Feb/2023:06:25:04 -0600] \"GET / HTTP/1.1\" 200 45","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.12","id":"200","url":"/"},"location":"/private/var/log/apache2/access_log"}

        ```
    5. Check the `Active response may not have an effect` alert is raised.
        ```JSON
        egrep "Active response may not" /var/ossec/logs/alerts/alerts.json
        
        {"timestamp":"2023-02-23T12:25:03.121+0000","rule":{"level":5,"description":"Active response related to pf has been activated, but may not have an effect because the firewall is inactive","id":"658","firedtimes":2,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677155103.45276","full_log":"2023/02/23 06:25:05 active-response/bin/firewall-drop: {\"message\":\"Active response may not have an effect\",\"profile\":\"default\",\"status\":\"inactive\",\"script\":\"pf\"}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"status":"inactive","message":"Active response may not have an effect","profile":"default","script":"pf"},"location":"/Library/Ossec/logs/active-responses.log"}

        ```

    </details>
</details>


<details><summary>Scenario 2: Firewall's file validation</summary>

  > <details><summary>Gherkin scenario description</summary>
  >
  >    ```Gherkin
  >    Given a wazuh-manager and a not started "<agent_os>" wazuh-agent.
  >    And the agent's firewal "<firewall_status>".
  >    And the file /etc/pf.conf is "<pf_conf_initial_status>".
  >    When start the wazuh-agent.
  >    Then the file /etc/pf.conf is "<pf_conf_final_status>".
  >  
  >    Examples:
  >    
  >    | agent_os| firewall_status | pf_conf_initial_status | pf_conf_final_status |
  >    |---------|-----------------|------------------------|----------------------|
  >    | Macos   | enabled         | not updated            | updated              |
  >    | Macos   | enabled         | updated                | not re-updated       |
  >    | Macos   | disabled        | not updated            | not updated          |
  >    | Macos   | disabled        | updated                | not re-updated       |
  >    | FreeBSD | enabled         | not updated            | updated              |
  >    | FreeBSD | enabled         | updated                | not re-updated       |
  >    | FreeBSD | disabled        | not updated            | not updated          |
  >    | FreeBSD | disabled        | updated                | not re-updated       |
  >    | OpenBSD | enabled         | not updated            | updated              |
  >    | OpenBSD | enabled         | updated                | not re-updated       |
  >    | OpenBSD | disabled        | not updated            | not updated          |
  >    | OpenBSD | disabled        | updated                | not re-updated       |
  >    ```
  >  </details>


  - <details><summary>Case 1 - Firewal enabled not updated file is updated</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`
    3. Check the file is updated with the wazuh rule.
        ```conf
        cat /etc/pf.conf
        
        ...
        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>Case 2 - Firewal disabled not updated file is not updated</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file is **not** updated with the wazuh rule.
        ```conf
        cat /etc/pf.conf

        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        ```
    </details>

  - <details><summary>Case 3 - Firewal enabled updated file is not re-updated</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file still has the wazuh rule.
        > This case is to check that the rule is not added twice
        ```conf
        cat /etc/pf.conf
        
        ...
        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>Case 4 - Firewal disabled updated file is not updated</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file still has the wazuh rule.
        > This case is to check the rule is not deleted or anything.
        ```conf
        cat /etc/pf.conf

        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

</details>

**Comment by QU3B1M** [comment]
## Update 27/02/2023
- After complications on the testing on FreeBSD and OpenBSD it was decided to exclude those cases as wazuh does not officially supports those kind of systems
- Started `upgrade` testing on Macos

> The complications on the BSD sytems seems to be because the images used, anyway multiple images were used and different issues were faced on each system, from not being able to stop/restart the firewall because the whole system crashes to an unexpected error when installing anything on the machine

**Comment by QU3B1M** [comment]
# Testing results :green_circle: 

## Unfixed

<details><summary>Error repoduction</summary>

1. Ensure the firewall is disabled on the `macos` system.
    ```powershell
    pfctl -d
    ```
2. Configure the POC environment from [this guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html).
3. "Attack" the macos system from an third system.
4. Check the `IP address found in AlienVault reputation database.` alert is raised.
    ```JSON
    egrep "IP address found" /var/ossec/logs/alerts/alerts.json

    {"timestamp":"2023-02-28T13:20:29.833+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":2,"mail":false,"groups":["attack"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677590429.17106","full_log":"192.168.56.12 - - [28/Feb/2023:07:20:28 -0600] \"GET / HTTP/1.1\" 200 45","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.12","id":"200","url":"/"},"location":"/private/var/log/apache2/access_log"}

    ```
5. The attacker IP is not blocked and there is no alert to inform why.
    ```
    curl http://192.168.56.11:8080
    
    <html><body><h1>It works!</h1></body></html>
    ```

</details>

## Upgrade

<details><summary>Upgrade process</summary>

- <details><summary>Upgrade `wazuh-agent`</summary>
  
    ```log

        Wazuh v4.5.0 (Rev. 40500) Installation Script - https://www.wazuh.com

        You are about to start the installation process of Wazuh.
        You must have a C compiler pre-installed in your system.

        - System: Darwin macos 18.7.0 (darwin 18.7)
        - User: root
        - Host: macos


        -- Press ENTER to continue or Ctrl-C to abort. --


        - You already have Wazuh installed. Do you want to update it? (y/n): y

            - Installation will be made at  /Library/Ossec .

        ...

        - Add the following lines to the beginning of your PF rules:
            More information at: 
            https://documentation.wazuh.com


            table <wazuh_fwtable> persist #wazuh_fwtable 
            block in quick from <wazuh_fwtable> to any
            block out quick from any to <wazuh_fwtable>


        - Update completed.

    ```
    </details>

- <details><summary>Upgrade `wazuh-manager`</summary>
  
    ```log

        Wazuh v4.5.0 (Rev. 40500) Installation Script - https://www.wazuh.com

        You are about to start the installation process of Wazuh.
        You must have a C compiler pre-installed in your system.

        - System: Darwin macos 18.7.0 (darwin 18.7)
        - User: root
        - Host: macos


        -- Press ENTER to continue or Ctrl-C to abort. --


        - You already have Wazuh installed. Do you want to update it? (y/n): y

            - Installation will be made at  /Library/Ossec .

        ...

        - Add the following lines to the beginning of your PF rules:
            More information at: 
            https://documentation.wazuh.com


            table <wazuh_fwtable> persist #wazuh_fwtable 
            block in quick from <wazuh_fwtable> to any
            block out quick from any to <wazuh_fwtable>


        - Update completed.

    ```
    </details>
</details>

<details><summary>Scenario 1: PoC environment and raised alert validation :green_circle:</summary>

  > <details><summary>Gherkin scenario description</summary>
  >
  >    ```Gherkin
  >    Given a wazuh-manager and a "<agent_os>" wazuh-agent.
  >    And the agent's firewal "<firewall_status>".
  >    And the environment configured as in the POC: https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html.
  >    When connect to the agent system from the attacker machine.
  >    Then the attacker's IP is "<IP_result>".
  >    And the POC alert is raised.
  >    
  >    And the alert "Active Response may not have an effect" alert is "<firewall_alert_status>"
  >  
  >    Examples:
  >    
  >    | agent_os| firewall_status | IP_result  | firewall_alert_status |
  >    |---------|-----------------|------------|-----------------------|
  >    | Macos   | enabled         | blocked    | not raised            |
  >    | Macos   | disabled        | not blocked| raised                |
  >    ```
  >  </details>

  - <details><summary>Case 1 - Macos firewall enabled IP is blocked :green_circle:</summary>
    
    1. Ensure the firewall is enabled on the `macos` system.
        ```powershell
        pfctl -e
        ```
    2. Configure the POC environment from [this guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html).
    3. "Attack" the macos system from an third system.
    4. Check the `IP address found in AlienVault reputation database.` alert is raised.
        ```JSON
        egrep "IP address found" /var/ossec/logs/alerts/alerts.json

        {"timestamp":"2023-02-28T13:35:20.995+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":5,"mail":false,"groups":["attack"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677591320.29591","full_log":"192.168.56.12 - - [28/Feb/2023:07:35:19 -0600] \"GET / HTTP/1.1\" 200 45","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.12","id":"200","url":"/"},"location":"/private/var/log/apache2/access_log"}
        ```
    5. Check the attacker's IP is blocked.
        ```
        curl http://192.168.56.11:8080
        
        ```
    6. Check the `Active response may not have an effect` alert is **not** raised.
        ```JSON
        egrep "Active response may not" /var/ossec/logs/alerts/alerts.json
        ```

    </details>


  - <details><summary>Case 2 - Macos firewall disabled IP is blocked :green_circle:</summary>
    
    1. Ensure the firewall is disabled on the `macos` system.
        ```powershell
        pfctl -d
        ```
    2. Configure the POC environment from [this guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html).
    3. "Attack" the macos system from an third system.
    4. Check the `IP address found in AlienVault reputation database.` alert is raised.
        ```JSON
        egrep "IP address found" /var/ossec/logs/alerts/alerts.json

        {"timestamp":"2023-02-28T13:20:29.833+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":2,"mail":false,"groups":["attack"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677590429.17106","full_log":"192.168.56.12 - - [28/Feb/2023:07:20:28 -0600] \"GET / HTTP/1.1\" 200 45","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.12","id":"200","url":"/"},"location":"/private/var/log/apache2/access_log"}

        ```
    5. Check the attacker's IP is **not** blocked.
        ```
        curl http://192.168.56.11:8080
        
        <html><body><h1>It works!</h1></body></html>
        ```
    
    6. Check the `Active response may not have an effect` alert is raised.
        ```JSON
        egrep "Active response may not" /var/ossec/logs/alerts/alerts.json
        
        {"timestamp":"2023-02-28T13:20:31.895+0000","rule":{"level":5,"description":"Active response related to pf has been activated, but may not have an effect because the firewall is inactive","id":"658","firedtimes":2,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677590431.19450","full_log":"2023/02/28 07:20:30 active-response/bin/firewall-drop: {\"message\":\"Active response may not have an effect\",\"profile\":\"default\",\"status\":\"inactive\",\"script\":\"pf\"}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"status":"inactive","message":"Active response may not have an effect","profile":"default","script":"pf"},"location":"/Library/Ossec/logs/active-responses.log"}

        ```

    </details>

</details>


<details><summary>Scenario 2: Firewall's file validation :green_circle:</summary>

  > <details><summary>Gherkin scenario description</summary>
  >
  >    ```Gherkin
  >    Given a wazuh-manager and a not started "<agent_os>" wazuh-agent.
  >    And the agent's firewal "<firewall_status>".
  >    And the file /etc/pf.conf is "<pf_conf_initial_status>".
  >    When start the wazuh-agent.
  >    Then the file /etc/pf.conf is "<pf_conf_final_status>".
  >  
  >    Examples:
  >    
  >    | agent_os| firewall_status | pf_conf_initial_status | pf_conf_final_status |
  >    |---------|-----------------|------------------------|----------------------|
  >    | Macos   | enabled         | not updated            | updated              |
  >    | Macos   | enabled         | updated                | not re-updated       |
  >    | Macos   | disabled        | not updated            | not updated          |
  >    | Macos   | disabled        | updated                | not re-updated       |
  >    ```
  >  </details>


  - <details><summary>Case 1 - Firewal enabled not updated file is updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`
    3. Check the file is updated with the wazuh rule.
        ```conf
        cat /etc/pf.conf
        
        ...
        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>Case 2 - Firewal disabled not updated file is not updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file is **not** updated with the wazuh rule.
        ```conf
        cat /etc/pf.conf

        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        ```
    </details>

  - <details><summary>Case 3 - Firewal enabled updated file is not re-updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file still has the wazuh rule.
        > This case is to check that the rule is not added twice
        ```conf
        cat /etc/pf.conf
        
        ...
        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>Case 4 - Firewal disabled updated file is not updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file still has the wazuh rule.
        > This case is to check the rule is not deleted or anything.
        ```conf
        cat /etc/pf.conf

        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

</details>




## Fresh install

<details><summary>Scenario 1: PoC environment and raised alert validation :green_circle:</summary>

  > <details><summary>Gherkin scenario description</summary>
  >
  >    ```Gherkin
  >    Given a wazuh-manager and a "<agent_os>" wazuh-agent.
  >    And the agent's firewal "<firewall_status>".
  >    And the environment configured as in the POC: https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html.
  >    When connect to the agent system from the attacker machine.
  >    Then the attacker's IP is "<IP_result>".
  >    And the POC alert is raised.
  >    
  >    And the alert "Active Response may not have an effect" alert is "<firewall_alert_status>"
  >  
  >    Examples:
  >    
  >    | agent_os| firewall_status | IP_result  | firewall_alert_status |
  >    |---------|-----------------|------------|-----------------------|
  >    | Macos   | enabled         | blocked    | not raised            |
  >    | Macos   | disabled        | not blocked| raised                |
  >    ```
  >  </details>

  - <details><summary>Case 1 - Macos firewall enabled IP is blocked :green_circle:</summary>
    
    1. Ensure the firewall is enabled on the `macos` system.
        ```powershell
        pfctl -e
        ```
    2. Configure the POC environment from [this guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html).
    3. "Attack" the macos system from an third system.
    4. Check the `IP address found in AlienVault reputation database.` alert is raised.
        ```JSON
        egrep "IP address found" /var/ossec/logs/alerts/alerts.json

        {"timestamp":"2023-02-23T12:06:58.165+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":1,"mail":false,"groups":["attack"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677154018.18950","full_log":"192.168.56.12 - - [23/Feb/2023:06:06:59 -0600] \"GET / HTTP/1.1\" 200 45","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.12","id":"200","url":"/"},"location":"/private/var/log/apache2/access_log"}
        ```
    5. Check the attacker's IP is blocked.
        ```
        curl http://192.168.56.11:8080
        
        ```
    6. Check the `Active response may not have an effect` alert is **not** raised.
        ```JSON
        egrep "Active response may not" /var/ossec/logs/alerts/alerts.json
        ```

    </details>

  - <details><summary>Case 2 - Macos firewall disabled IP is blocked :green_circle:</summary>
    
    1. Ensure the firewall is disabled on the `macos` system.
        ```powershell
        pfctl -d
        ```
    2. Configure the POC environment from [this guide](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html).
    3. "Attack" the macos system from an third system.
    4. Check the `IP address found in AlienVault reputation database.` alert is raised.
        ```JSON
        egrep "IP address found" /var/ossec/logs/alerts/alerts.json

        {"timestamp":"2023-02-23T12:25:02.813+0000","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":7,"mail":false,"groups":["attack"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677155102.42932","full_log":"192.168.56.12 - - [23/Feb/2023:06:25:04 -0600] \"GET / HTTP/1.1\" 200 45","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.12","id":"200","url":"/"},"location":"/private/var/log/apache2/access_log"}

        ```
    5. Check the attacker's IP is **not** blocked.
        ```
        curl http://192.168.56.11:8080
        
        <html><body><h1>It works!</h1></body></html>
        ```
    
    6. Check the `Active response may not have an effect` alert is raised.
        ```JSON
        egrep "Active response may not" /var/ossec/logs/alerts/alerts.json
        
        {"timestamp":"2023-02-23T12:25:03.121+0000","rule":{"level":5,"description":"Active response related to pf has been activated, but may not have an effect because the firewall is inactive","id":"658","firedtimes":2,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"001","name":"macos","ip":"10.0.2.15"},"manager":{"name":"manager"},"id":"1677155103.45276","full_log":"2023/02/23 06:25:05 active-response/bin/firewall-drop: {\"message\":\"Active response may not have an effect\",\"profile\":\"default\",\"status\":\"inactive\",\"script\":\"pf\"}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"status":"inactive","message":"Active response may not have an effect","profile":"default","script":"pf"},"location":"/Library/Ossec/logs/active-responses.log"}

        ```

    </details>

</details>


<details><summary>Scenario 2: Firewall's file validation :green_circle:</summary>

  > <details><summary>Gherkin scenario description</summary>
  >
  >    ```Gherkin
  >    Given a wazuh-manager and a not started "<agent_os>" wazuh-agent.
  >    And the agent's firewal "<firewall_status>".
  >    And the file /etc/pf.conf is "<pf_conf_initial_status>".
  >    When start the wazuh-agent.
  >    Then the file /etc/pf.conf is "<pf_conf_final_status>".
  >  
  >    Examples:
  >    
  >    | agent_os| firewall_status | pf_conf_initial_status | pf_conf_final_status |
  >    |---------|-----------------|------------------------|----------------------|
  >    | Macos   | enabled         | not updated            | updated              |
  >    | Macos   | enabled         | updated                | not re-updated       |
  >    | Macos   | disabled        | not updated            | not updated          |
  >    | Macos   | disabled        | updated                | not re-updated       |
  >    ```
  >  </details>


  - <details><summary>Case 1 - Firewal enabled not updated file is updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`
    3. Check the file is updated with the wazuh rule.
        ```conf
        cat /etc/pf.conf
        
        ...
        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>Case 2 - Firewal disabled not updated file is not updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file is **not** updated with the wazuh rule.
        ```conf
        cat /etc/pf.conf

        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        ```
    </details>

  - <details><summary>Case 3 - Firewal enabled updated file is not re-updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file still has the wazuh rule.
        > This case is to check that the rule is not added twice
        ```conf
        cat /etc/pf.conf
        
        ...
        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

  - <details><summary>Case 4 - Firewal disabled updated file is not updated :green_circle:</summary>

    1. Ensure the firewall is enabled on the `macos` system.
    2. Start the macos `wazuh-agent`.
    3. Check the file still has the wazuh rule.
        > This case is to check the rule is not deleted or anything.
        ```conf
        cat /etc/pf.conf

        dummynet-anchor "com.apple/*"
        anchor "com.apple/*"
        load anchor "com.apple" from "/etc/pf.anchors/com.apple"
        table <wazuh_fwtable> persist #wazuh_fwtable
        block in quick from <wazuh_fwtable> to any
        block out quick from any to <wazuh_fwtable>
        ```
    </details>

</details>

**Comment by damarisg** [comment]
## QA review
- **Type**: Manual Testing
- **Status**: Approved  🟢 
- **Comments**: Everything seems to work correctly.
